File: ignite-nest/src/comment/infrastructure/web/dto/response/comment-list.dto.ts
import { PickType } from '@nestjs/swagger';
import { Comment } from '../../../../entity/comment.entity';
import { PostListDto } from '../../../../../post/infrastructure/web/dto/response/post-list.dto';

export class CommentListDto extends PickType(Comment, [
  'id',
  'title',
  'content',
  'createdAt',
]) {
  post: PostListDto;

  constructor(
    id: number,
    title: string,
    content: string,
    createdAt: Date,
    postListDto: PostListDto,
  ) {
    super(id, title, content, createdAt);
    this.post = postListDto;
  }
}

File: ignite-spring/src/main/kotlin/org/rookedsysc/ignitespring/comment/infrastructure/web/dto/CommentListDto.kt
package org.rookedsysc.ignitespring.comment.infrastructure.web.dto

import org.rookedsysc.ignitespring.post.infrastructure.web.dto.PostListDto
import java.util.Date

data class CommentListDto(
    val id: Long,
    val title: String,
    val content: String,
    val createdAt: Date,
    val postListDto: PostListDto
) {}
